opentelemetry-user-events-logs/src/logs/processor.rs

```diff
@@ -93,7 +96,47 @@ impl<'a> ProcessorBuilder<'a> {
     /// For example the following will capture level 2 (Error) and 3(Warning) events:
     /// perf record -e user_events:myprovider_L2K1,user_events:myprovider_L3K1
     pub(crate) fn new(provider_name: &'a str) -> Self {
-        Self { provider_name }
+        Self {
+            provider_name,
+            resource_attribute_keys: HashSet::new(),
+        }
+    }
+
+    /// Sets the resource attributes for the processor.
+    ///
+    /// This specifies which resource attributes should be exported with each log record.
+    ///
+    /// # Performance Considerations
+    ///
+    /// **Warning**: Each specified resource attribute will be serialized and sent
+    /// with EVERY log record. This is different from OTLP exporters where resource
+    /// attributes are serialized once per batch. Consider the performance impact
+    /// when selecting which attributes to export.
+    ///
+    /// # Best Practices for user_events
+    ///
+    /// **Recommendation**: Be selective about which resource attributes to export.
+    /// Since user_events writes to a local kernel buffer and requires a local
+    /// listener/agent, the agent can often deduce many resource attributes without
+    /// requiring them to be sent with each log:
+    ///
+    /// - **Infrastructure attributes** (datacenter, region, availability zone) can
+    ///   be determined by the local agent
+    /// - **Host attributes** (hostname, IP address, OS version) are available locally
+    /// - **Deployment attributes** (environment, cluster) may be known to the agent
+    ///
+    /// Focus on attributes that are truly specific to your application instance
+    /// and cannot be easily determined by the local agent.
+    pub fn with_resource_attributes<I, S>(mut self, attributes: I) -> Self
+    where
+        I: IntoIterator<Item = S>,
+        S: Into<Cow<'static, str>>,
+    {
+        self.resource_attribute_keys = attributes
+            .into_iter()
+            .map(|s| s.into().into_owned())
```

Comment:
This negates the value of using `Cow<'static, str>`. If we want to avoid allocations for static strings, we need to use `Cow<static, str>` throughout the processor builder and exporter. If not, we could have simply used `S: Into<String>`.